`mypy` 1.20.2 fails in CI due to 3.12 syntax in a dependency

```py
Run mypy --version && mypy --config-file examples/mypy.cfg examples/basic/scatters/color_scatter.py
mypy 1.20.2 (compiled: yes)
/home/runner/miniconda3/envs/bk-test/lib/python3.14/site-packages/sphinx/ext/autodoc/_property_types.py:30: error: Type statement is only supported in Python 3.12 and greater  [syntax]
Found 1 error in 1 file (errors prevented further checking)
Error: Process completed with exit code 2.
```

--- a/examples/mypy.cfg
+++ b/examples/mypy.cfg
@@ -1,3 +1,8 @@
 [mypy]
 python_version = 3.11
 strict = true
+
+[mypy-sphinx]
+follow_imports = skip
+[mypy-sphinx.*]
+follow_imports = skip

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -297,5 +297,5 @@ ignore_errors = false
 
 # TODO remove when we drop Python 3.11
 [[tool.mypy.overrides]]
-module = ["sphinx.*"]
+module = ["sphinx", "sphinx.*"]
 follow_imports = "skip"

